### DISCUSSION TITLE
Install skill for development

### DISCUSSION THREAD

**Discussion by mvrius** [OpenVoiceOS/ovos-buildroot#269] [Q&A] [answered]
Hi,

Trying to build a new skill, and noticing that if I install the skill, say the hello world example, with "pip install -e ." ; then it cannot be loaded.

```
2024-06-06 23:02:07.137 - skills - ovos_plugin_manager.utils:find_plugins:124 
ERROR - Failed to load plugin entry point EntryPoint(name='skill-ovos-hello-world.openvoiceos', 
value='skill_ovos_hello_world:HelloWorldSkill', group='ovos.plugin.skill'): No module named 'skill_ovos_hello_world'
```

If I bdist_wheel build the skill and then install it, then everything works as expected.

How do I install a skill during development to streamline change propagation?

Thanks!

**Comment by JarbasAl** [comment, marked as answer]
you can either use the old mycroft way, where you just git clone a skill into the folder and edit it live, then it should restart when changes are detected (no installing)

or use ovos-workshop, you can [launch a skill standalone](https://github.com/OpenVoiceOS/OVOS-workshop/blob/dev/setup.py#L80) in it's own process and restart as often as you need during development. This is similar to how skills are deployed in [their own docker containers](https://github.com/OpenVoiceOS/ovos-docker/blob/dev/skills/skill-date-time/Dockerfile#L25)
